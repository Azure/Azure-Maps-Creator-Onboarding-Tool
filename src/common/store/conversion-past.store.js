import { create } from 'zustand';

import { conversionStatuses, conversionSteps} from './conversion.store';

export const useConversionPastStore = create((set) => ({
  selectedStep: conversionSteps.upload,
  uploadStepStatus: conversionStatuses.empty,
  uploadUdId: null,
  conversionStepStatus: conversionStatuses.empty,
  conversionId: null,
  datasetStepStatus: conversionStatuses.empty,
  datasetId: null,
  tilesetStepStatus: conversionStatuses.empty,
  tilesetId: null,
  mapConfigurationId: null,
  bbox: null,
  setStep: (selectedStep) => set({
    selectedStep,
  }),
  setData: (data) => set(data),
}));