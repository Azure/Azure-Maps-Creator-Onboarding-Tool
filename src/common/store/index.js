export { conversionSteps, useConversionStore } from './conversion.store';
export { useDissolvedExterior, useGeometryStore } from './geometry.store';
export { useLayersStore } from './layers.store';
export { useLevelsStore } from './levels.store';
export {
  progressBarStepsByKey,
  useCompletedSteps,
  useProgressBarSteps,
  useProgressBarStore,
} from './progress-bar-steps';
export { resetStores } from './reset';
export { LRO_STATUS, useResponseStore } from './response.store';
export { useReviewManifestJson, useReviewManifestStore } from './review-manifest.store';
export { getDomain, useUserStore } from './user.store';
