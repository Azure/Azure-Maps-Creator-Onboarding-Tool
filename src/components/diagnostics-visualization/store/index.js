import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from '../reducers';

// Middlewares go here
const middleware = {};

const configureStore = (initialState) => createStore(
  rootReducer,
  initialState,
  // Redux Dev Tools Middleware
  composeWithDevTools(applyMiddleware(...middleware)),
);

// pass an optional param to rehydrate state on app start
const store = configureStore();

// export store singleton instance
export default store;
