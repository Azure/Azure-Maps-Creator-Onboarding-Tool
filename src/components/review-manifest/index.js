import ReviewManifest from './review-manifest';

export default ReviewManifest;