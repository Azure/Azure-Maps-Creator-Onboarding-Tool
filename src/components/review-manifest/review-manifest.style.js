import { css } from '@emotion/css';

import { color, fontSize } from 'common/styles';

export const reviewManifestFooter = css`
  border-top: 1px solid ${color.darkGray};
  padding: 1rem 1.25rem;
  background: white;
`;

export const buttonStyle = css`
  margin-right: 0.625rem;
  height: 1.5rem;
  font-size: ${fontSize.sm};
`;

export const panelStyle = {
  commands: {
    padding: 0,
    height: 50,
    alignItems: 'center',
    display: 'flex',
    width: '100%',
    position: 'absolute!important',
    top: '0px!important',
    background: 'white',
  },
  content: {
    // 50px and 73px here at top and bottom are heights of header and footer+padding respectively
    padding: '50px 24px 73px',
  },
  footer: {
    position: 'absolute!important',
    width: '100%',
    bottom: '0px!important',
  },
  footerInner: {
    padding: 0,
    height: 57,
  },
  navigation: {
    width: '100%',
  }
};