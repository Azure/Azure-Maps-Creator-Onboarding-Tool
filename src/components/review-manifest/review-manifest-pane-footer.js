import { useCallback } from 'react';
import { DefaultButton, PrimaryButton } from '@fluentui/react';
import { useTranslation } from 'react-i18next';
import { saveAs } from 'file-saver';

import { useReviewManifestJson, useReviewManifestStore } from 'common/store/review-manifest.store';
import { buttonStyle, reviewManifestFooter } from './review-manifest.style';

const reviewManifestSelector = (s) => s.hidePane;
export const filename = 'manifest.json';
export const fileType = 'application/json';

const Footer = () => {
  const hidePane = useReviewManifestStore(reviewManifestSelector);
  const json = useReviewManifestJson();
  const { t } = useTranslation();

  const saveFile = useCallback(() => {
    saveAs(
      new Blob([JSON.stringify(json, null, 2)], {
        type: fileType,
      }),
      filename,
    );
  }, [json]);

  return (
    <div className={reviewManifestFooter}>
      <PrimaryButton className={buttonStyle} onClick={saveFile} data-test-id='download-manifest-button'>
        {t('download')}
      </PrimaryButton>
      <DefaultButton className={buttonStyle} onClick={hidePane}>
        {t('cancel')}
      </DefaultButton>
    </div>
  );
};

export default Footer;