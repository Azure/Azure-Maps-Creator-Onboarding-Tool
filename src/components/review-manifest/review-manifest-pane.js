import { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import ReactJson from 'react-json-view';
import { Panel, PanelType } from '@fluentui/react/lib/Panel';

import { useReviewManifestJson, useReviewManifestStore } from 'common/store/review-manifest.store';
import { panelStyle } from './review-manifest.style';
import Footer from './review-manifest-pane-footer';

const reviewManifestSelector = (s) => s.hidePane;

const ReviewManifestPane = () => {
  const hidePane = useReviewManifestStore(reviewManifestSelector);
  const json = useReviewManifestJson();
  const { t } = useTranslation();

  const onRenderFooterContent = useCallback(
    () => <Footer />,
    [],
  );

  return (
    <Panel headerText={t('review.manifest')} isLightDismiss isOpen={true} onDismiss={hidePane}
           closeButtonAriaLabel={t('close')} type={PanelType.medium} styles={panelStyle}
           onRenderFooterContent={onRenderFooterContent} isFooterAtBottom={true}>
      <ReactJson src={json} iconStyle='square' indentWidth={2} displayDataTypes={false} name={false}
                 displayObjectSize={false} enableClipboard={false} displayArrayKey={false} />
    </Panel>
  );
};

export default ReviewManifestPane;