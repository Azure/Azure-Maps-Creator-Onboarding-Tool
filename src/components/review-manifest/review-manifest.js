import { useReviewManifestStore } from 'common/store/review-manifest.store';

import ReviewManifestPane from './review-manifest-pane';

const reviewManifestSelector = (s) => s.isPaneShown;

const ReviewManifest = () => {
  const isPaneShown = useReviewManifestStore(reviewManifestSelector);

  if (!isPaneShown) {
    return null;
  }

  return <ReviewManifestPane />;
};

export default ReviewManifest;