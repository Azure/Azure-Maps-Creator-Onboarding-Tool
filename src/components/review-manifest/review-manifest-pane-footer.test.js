import { fireEvent, render, screen } from '@testing-library/react';
import { saveAs } from 'file-saver';

import { useReviewManifestStore } from 'common/store/review-manifest.store';
import ReviewManifestPaneFooter, { filename } from './review-manifest-pane-footer';

jest.mock('file-saver', () => ({
  saveAs: jest.fn(),
}));

describe('ReviewManifestPane Footer', () => {
  it('should render review manifest pane footer', () => {
    const view = render(<ReviewManifestPaneFooter />);
    expect(view).toMatchSnapshot();
  });

  it('should close the pane when cancel button is clicked', () => {
    const state = useReviewManifestStore.getState();
    const spy = jest.spyOn(state, 'hidePane');
    render(<ReviewManifestPaneFooter />);
    const cancelBtn = screen.getByText('cancel');
    fireEvent.click(cancelBtn);
    expect(spy).toHaveBeenCalled();
  });

  it('should save file when download button is clicked', () => {
    render(<ReviewManifestPaneFooter />);
    const cancelBtn = screen.getByText('download');
    fireEvent.click(cancelBtn);
    expect(saveAs).toHaveBeenCalledWith(new Blob(), filename);
  });
});