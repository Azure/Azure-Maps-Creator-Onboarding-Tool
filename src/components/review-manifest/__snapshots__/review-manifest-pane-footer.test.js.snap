// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewManifestPane Footer should render review manifest pane footer 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  border-top: 1px solid #ccc;
  padding: 1rem 1.25rem;
  background: white;
}

.emotion-1 {
  margin-right: 0.625rem;
  height: 1.5rem;
  font-size: 0.813rem;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <button
          class="ms-Button ms-Button--primary emotion-1 root-109"
          data-is-focusable="true"
          data-test-id="download-manifest-button"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <span
              class="ms-Button-textContainer textContainer-111"
            >
              <span
                class="ms-Button-label label-113"
                id="id__0"
              >
                download
              </span>
            </span>
          </span>
        </button>
        <button
          class="ms-Button ms-Button--default emotion-1 root-117"
          data-is-focusable="true"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-110"
            data-automationid="splitbuttonprimary"
          >
            <span
              class="ms-Button-textContainer textContainer-111"
            >
              <span
                class="ms-Button-label label-113"
                id="id__3"
              >
                cancel
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  border-top: 1px solid #ccc;
  padding: 1rem 1.25rem;
  background: white;
}

.emotion-1 {
  margin-right: 0.625rem;
  height: 1.5rem;
  font-size: 0.813rem;
}

<div>
    <div
      class="emotion-0"
    >
      <button
        class="ms-Button ms-Button--primary emotion-1 root-109"
        data-is-focusable="true"
        data-test-id="download-manifest-button"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <span
            class="ms-Button-textContainer textContainer-111"
          >
            <span
              class="ms-Button-label label-113"
              id="id__0"
            >
              download
            </span>
          </span>
        </span>
      </button>
      <button
        class="ms-Button ms-Button--default emotion-1 root-117"
        data-is-focusable="true"
        type="button"
      >
        <span
          class="ms-Button-flexContainer flexContainer-110"
          data-automationid="splitbuttonprimary"
        >
          <span
            class="ms-Button-textContainer textContainer-111"
          >
            <span
              class="ms-Button-label label-113"
              id="id__3"
            >
              cancel
            </span>
          </span>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
